python/nammu/utils/__init__.py

```diff
@@ -122,7 +123,12 @@ def get_yaml_config(yaml_filename):
     # In Unix getResource returns the path with prefix "file:" but in
     # Windows prefix is "jar:file:"
     path_to_jar = str(config_file_url).split('file:')[1]
-    path_to_jar = path_to_jar.split('!')[0]
+    # The path will be of the form /path/to/jar!path/inside/jar
+    # Take everything up to the final !, in case previous parts also contain !
+    path_to_jar = path_to_jar.rsplit('!', 1)[0]
+    # Decode any special characters contained in the path so that, for example,
+    # we use 'dir name' instead of 'dir%20name'
+    path_to_jar = urllib.unquote(path_to_jar)
```

Comment:
This looks good. My only concern is that we are manually splitting and joining paths.

We should be using `os.path` methods to do this in a platform independent way. I think this fix should be fine as it does not make any assumptions about forward or back slashes. I'm going to open a new issue so that when we next do some refactoring we can look at addressing this across the whole of the codebase.

Reply:
Agreed about using `os.path` for path manipulations! I tried to avoid any more splitting than what was already done in the previous version.